### DISCUSSION TITLE
Customize error.log with NGINX $request_id

### DISCUSSION THREAD

**Discussion by shawnchapla-8451** [Kong/kong#11835] [Help] [answered]
Wondering if anyone has found a way to customize each line written to Kong's error log such that it allows you to associate the information in there with its corresponding access.log entry? For example, if you look at a sample entry written while debug logging is enabled:
```
2023/10/23 13:39:06 [debug] 1133905#0: *658 [lua] init.lua:1127: balancer(): setting address (try 1): <IP redacted>:443
```

I see [here](https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.log/#konglog) the description of the part of the log entry that starts with ```[lua]``` in the example, so I assume anything to the left of that is unchangeable.

What I'm trying to do is somehow add the value of the $request_id variable to what gets written to error.log, in order to create that correlation between the two logs; that you can find all the error.log entries that go with a particular entry in access.log.

**Comment by chronolaw** [comment, marked as answer]
We have introduced this feature, see: https://github.com/Kong/kong/pull/11663.
Notice, there is another PR changing the variable to `$kong_request_id`. (https://github.com/Kong/kong/pull/11725)

**Reply by shawnchapla-8451** [reply]
Thanks for this.  This change looks quite recent ... how long does it typically take before it's part of a new released version of Kong?

**Reply by guanlan** [reply]
@shawnchapla-8451 The release has targeted within the next few weeks.
